server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -368,7 +368,7 @@ public DesiredBalance compute(
         }
 
         long lastConvergedIndex = hasChanges ? previousDesiredBalance.lastConvergedIndex() : desiredBalanceInput.index();
-        return new DesiredBalance(lastConvergedIndex, assignments);
+        return new DesiredBalance(lastConvergedIndex, assignments, delegateAllocator.getNodeWeightStats());
```

Comment:
I wonder whether we are interested in how many iterations it takes to get the desired balance and whether it is interupted due to things like new state?

Reply:
+1. We might replicate some of the org.elasticsearch.cluster.routing.allocation.allocator.DesiredBalanceStats stats as APM metrics. Additionally it would be nice to have a metric representing the fraction of reconciliations resulted in an allocation same to desired one (in other words a metric representing if cluster can keep up with computed balances).

As commented in https://github.com/elastic/elasticsearch/pull/115854/files#r1822994367, this might be part of a different issue and a separate pr.

Reply:
yes, those are also interesting. there are a lot of metrics we can expose about the allocator. but not all in the same PR! :) What you're asking for seems to be part of ES-9872.

Reply:
Ah I was not aware these computer level metrics are already exposed via the allocatorStats. Thanks for explaining. 👍